Tests/Unit/NetworkServer/LoRaDeviceCacheTest.cs

```diff
@@ -381,6 +384,20 @@ public TestDeviceCache(Action<LoRaDevice> onRefreshDevice)
             internal async Task WaitForRefreshAsync(CancellationToken cancellationToken) =>
                 await this.refreshTick.WaitAsync(cancellationToken);
 
+            internal async Task WaitForRefreshCallsAsync(int numberOfCalls, TimeSpan timeout = default)
```

Comment:
Maybe this might be a candidate for reusing `MoqExtensions.RetryVerifyAsync`?

Reply:
I would agree if we didn't had the explicit support in the test class to verify for the refreshes. The semaphore signaling makes it more deterministic. If the machine is not overloaded, the timeouts won't kick in and we have a quicker test run.